src/compiler/crystal/syntax/ast.cr

```diff
@@ -1351,7 +1351,9 @@ module Crystal
   end
 
   class Generic < ASTNode
-    property name : Path
+    # Usually a Path, but can also be a TypeNode in the case of a
+    # custom array/hash-like literal.
+    property name : ASTNode
```

Comment:
I know it has already been merged, but why not restrict on `TypeNode | Path` instead?

Reply:
Because the compiler will find the common ancestor, which is `ASTNode`. It's an ugly behavior of the language but that's how things work...

Reply:
Right, I forgot about that, thank you